types/trace.go

```diff
@@ -0,0 +1,132 @@
+package types
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+)
+
+// TraceSession model.
+type TraceSession struct {
+	ID         string    `json:"id" yaml:"id"`
+	PipelineID string    `json:"pipelineID" yaml:"pipelineID"`
+	Plugins    []string  `json:"plugins" yaml:"plugins"`
+	Lifespan   Duration  `json:"lifespan" yaml:"lifespan"`
+	CreatedAt  time.Time `json:"createdAt" yaml:"createdAt"`
+	UpdatedAt  time.Time `json:"updatedAt" yaml:"updatedAt"`
+}
+
+// Active tells whether a session is still within its lifespan.
+func (ts TraceSession) Active() bool {
+	return ts.CreatedAt.Add(time.Duration(ts.Lifespan)).After(time.Now())
+}
+
+// CreateTraceSession request payload for creating a new trace session.
+type CreateTraceSession struct {
+	Plugins  []string `json:"plugins"`
+	Lifespan Duration `json:"lifespan"`
+}
+
+// TraceSessionsParams request payload for querying trace sessions.
+type TraceSessionsParams struct {
+	Last   *uint64
+	Before *string
+}
+
+// TraceSessions paginated list.
+type TraceSessions struct {
+	Items     []TraceSession `json:"items"`
+	EndCursor *string        `json:"endCursor"`
+}
+
+// UpdateTraceSession request payload for updating a trace session.
+type UpdateTraceSession struct {
+	Plugins  *[]string `json:"plugins"`
+	Lifespan *Duration `json:"lifespan"`
+}
+
+// TraceRecord model.
+type TraceRecord struct {
+	ID        string    `json:"id" yaml:"id"`
+	SessionID string    `json:"sessionID" yaml:"sessionID"`
+	CreatedAt time.Time `json:"createdAt" yaml:"createdAt"`
+
+	// fluent-bit data from here on.
+	Kind           TraceRecordKind `json:"type" yaml:"type"`
+	TraceID        string          `json:"traceID" yaml:"traceID"`
```

Comment:
The field is called trace_id in the JSON protocol.

Reply:
👍 
I found that strange, since all others were using snake_case. Thanks.

Reply:
Fixed.